### PR TITLE
Add mechanisms to reformat and check code style, and reformat whole codebase (#128)

### PR DESCRIPTION
**Pull request by rouault**
Use an internal version of astyle (astyle 3.0). Scripts taken from QGIS.
astyle.options from https://github.com/uclouvain/openjpeg/issues/128
    
scripts/prepare-commit.sh can be used locally to automatically reformat edited files.
    
Travis-CI will run scripts/verify-indentation.sh to verify committed files.

### CODE REVIEW

**Comment by rouault** [comment]
Example of (voluntary) failure of a Travis job due to syntax error: https://travis-ci.org/rouault/openjpeg/jobs/230491462

**Comment by stweil** [comment]
While adding GPL code to `thirdparty` might be considered to be legally fine, I don't think that adding such code to `scripts` is a good idea.

**Comment by rouault** [comment]
@detonin If you intend to do a release, I guess you might want to make it before I merge this.

**Comment by rouault** [comment]
> While adding GPL code to thirdparty might be considered to be legally fine,
Actually the astyle source code is a X/MIT licence : https://github.com/rouault/openjpeg/blob/d4e54e9f35d532062533f1d369c159810b01d224/thirdparty/astyle/LICENSE.md

> I don't think that adding such code to scripts is a good idea.
Someone could possibly later rewrite them if they find that the investment is worth, but my shell abilities are too poor and this would be wheel reinvention. Anyway anybody can safely strip the scripts/ directory for a tarball and still successfully build openjpeg if they are so afraid of GPL.

**Comment by boxerab** [comment]
@stweil packaging BSD and GPL files together doesn't trigger the GPL on the BSD project. It is only when the BSD proj is linked to the GPL that the GPL license is triggered. So, there shouldn't be any legal issues with putting a GPL script in the project.

File: scripts/astyle.options [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--convert-tabs
+--lineend=linux
+--indent=spaces=4
+--style=kr
+--add-brackets
```
Comment (detonin) [inline scripts/astyle.options]: should be --add-braces (--add-brackets is deprecated)

File: scripts/astyle.options [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+--max-code-length=80
+--break-after-logical
+--pad-header
+--pad-oper
+--unpad-paren
+--suffix=none
```
Comment (detonin) [inline scripts/astyle.options]: @rouault Where does theses options come from ? Are they copy-pasted from QGIS ?
@malaterre @mayeut Any comment on the astyle config before we merge the PR ?

Comment (rouault) [inline scripts/astyle.options]: All those options come from https://github.com/uclouvain/openjpeg/issues/128#issuecomment-175076907

File: scripts/astyle.options (hunk 1 above)
Comment (rouault) [inline scripts/astyle.options]: Fixed

File: .travis.yml [hunk 3]
```diff
@@ -1,25 +1,37 @@
-language: c
+language: cpp
```
Comment (detonin) [inline .travis.yml]: Is that required ? Changing the whole project from C to C++ (also in cmake) just for opjstyle seems a bit overkill. No other option ?

Comment (rouault) [inline .travis.yml]: I found hard with Travis to make it pass on all configurations and I'm afraid I didn't find a better solution.

Comment (detonin) [inline .travis.yml]: Are there any implication of changing compiler from gcc to g++ ? Does this prevent to compile on some older platforms ? 
@malaterre any advice on this ?

**Comment by detonin** [comment]
> @detonin If you intend to do a release, I guess you might want to make it before I merge this.

I'll rather do the release afterwards. But before merging I would like some feedback on the compiler change.

@boxerab thanks for your comment on licensing

File: .travis.yml (hunk 3 above)
Comment (rouault) [inline .travis.yml]: cmake will require a c++ compiler to be detected, even if it doesn't use it (it is only used if WITH_ASTYLE is ON, which is not the default). There's perhaps a way to restrict c++ compiler detection only when WITH_ASTYLE is ON but I couldn't find it.
The C compiler (gcc) will still be used to compile the .c files.

Anyway we require CMake to be able to build openjpeg and CMake source code itself is C++, so the platform must have a working C++ compiler.

And at some point we might perhaps want to use C++ for openjpeg itself.

Comment (malaterre) [inline .travis.yml]: @rouault you are describing the symptoms not the real issue :) The fact that cmake only works on arch with valid c++ compiler is outside the scope of this question. I also fail to understand why `WITH-ASTYLE=ON` imply a valid c++ compiler in the PATH. I do understand that it should have zero impact on the build itself (a bit more dependencies though).

Comment (rouault) [inline .travis.yml]: @malaterre Astyle source code is C++ and apparently the fact that a project is c, c++ or both is declared in cmake at the PROJECT() scope, so as soon as you have C++ code you must enable C++. Is it really an issue to require a C++ compiler available ? I don't think so honestly.

Comment (malaterre) [inline .travis.yml]: I totally failed to follow the fact that now openjpeg is bootstraping a c++ project (astyle). In that case, yes there isn't much solution. As said above `it should have zero impact in the build itself`, I was just curious.

Comment (detonin) [inline .travis.yml]: ok thanks for your feedback @malaterre .

@rouault I think we can proceed and merge now.

**Comment by rouault** [comment]
@detonin Reformat source code PR merged

**Comment by rouault** [comment]
@detonin Wait: I see the .h files were not reformatted. Addressed by https://github.com/uclouvain/openjpeg/pull/926

**Comment by rouault** [comment]
@detonin OK, #926 merged. Things should be good now

**Pull request merged**